src/features/defi/providers/fox-eth-lp/components/FoxEthLpManager/Overview/FoxEthLpOverview.tsx

```diff
@@ -69,19 +103,13 @@ export const FoxEthLpOverview: React.FC<FoxEthLpOverviewProps> = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [highestBalanceAccountId])
 
-  const { underlyingFoxAmount, underlyingEthAmount } = opportunity!
-
-  const lpAsset = useAppSelector(state => selectAssetById(state, opportunity?.assetId ?? ''))
-  const foxAsset = useAppSelector(state => selectAssetById(state, foxAssetId))
-  const ethAsset = useAppSelector(state => selectAssetById(state, ethAssetId))
-
   const selectedLocale = useAppSelector(selectSelectedLocale)
   const descriptionQuery = useGetAssetDescriptionQuery({
     assetId: lpAsset?.assetId,
     selectedLocale,
   })
 
-  if (!lpAsset || !opportunity || !opportunity.isLoaded) {
+  if (!opportunityMetadata || !underlyingAssetsWithBalances) {
```

Comment:
That's all we basically need to check now, if we have metadata and balances then we're 🛍️